bin/metrics-vsphere.rb

```diff
@@ -124,46 +159,59 @@ def find_or_take_first(resources, resource_name)
-
-    pm = vim.serviceInstance.content.perfManager
 
-    compute_resources = dc.hostFolder.children.grep(RbVmomi::VIM::ComputeResource)
-    if config[:vm_name] || config[:host_name] || config[:compute_resource]
-      compute_resource = find_or_take_first(compute_resources, config[:compute_resource])
-    end
-    host = find_or_take_first(compute_resource.host, config[:host_name]) if config[:vm_name] || config[:host_name]
-
-    if config[:vm_name]
-      vms = host.vm.grep(RbVmomi::VIM::VirtualMachine)
-      resource = find_or_take_first(vms, config[:vm_name])
-    elsif config[:host_name]
-      resource = host
-    elsif config[:compute_resource] && compute_resource.is_a?(RbVmomi::VIM::ClusterComputeResource)
-      resource = compute_resource
+    if config[:find_resource]
+      result = find_resource_by_name(build_tree_resources(data_centers), config[:find_resource])
+      if result.empty?
+        puts 'Not found'
+      else
+        result.each do |line|
+          puts line
+        end
+      end
+      ok
     else
-      resource = dc
-    end
+      dc = find_or_take_first(data_centers, config[:data_center])
+      pm = vim.serviceInstance.content.perfManager
 
-    regexp = Regexp.new("^#{config[:command_type]}")
-    stats_hash = {
-      multi_instance: true,
-      interval: config[:period],
-      start_time: (Time.now - config[:period])
-    }
-    metrics = pm.retrieve_stats([resource],
-                                [],
-                                stats_hash)
-
-    if metrics
-      filtered_metrics = metrics[resource][:metrics].select { |(metric, _), _| metric.to_s.match(regexp) && metric }
-      filtered_metrics.each do |(metric, instance), value|
-        output "#{config[:scheme]}.#{[instance, metric].flatten.reject { |e| e == '' }.join('.')}", value.first
+      compute_resources = dc.hostFolder.children.grep(RbVmomi::VIM::ComputeResource)
+      if config[:vm_name] || config[:host_name] || config[:compute_resource]
+        compute_resource = find_or_take_first(compute_resources, config[:compute_resource])
+      end
+      host = find_or_take_first(compute_resource.host, config[:host_name]) if config[:vm_name] || config[:host_name]
+
+      if config[:vm_name]
+        vms = host.vm.grep(RbVmomi::VIM::VirtualMachine)
+        resource = find_or_take_first(vms, config[:vm_name])
+      elsif config[:host_name]
+        resource = host
+      elsif config[:compute_resource] && compute_resource.is_a?(RbVmomi::VIM::ClusterComputeResource)
+        resource = compute_resource
+      else
+        resource = dc
       end
 
-      ok
-    else
-      warning
+      regexp = Regexp.new("^#{config[:command_type]}")
+      metrics = pm.retrieve_stats([resource],
+                                  [],
+                                  multi_instance: true, interval: config[:period],
+                                  start_time: (Time.now - config[:period] * 10))
```

Comment:
what is the `* 10` accounting for?

Reply:
Have you attempted to sync time with NTP or Chrony? Here is what I found as far as steps from a [VMware perspective](https://kb.vmware.com/s/article/2012069). For your laptop if it is out of sync from the world depending on your OS (and potentially distro) there are different steps you can take. If you can let me know which os/distro you are using I can send you more specific instructions. I don't feel like accounting for a time lag in the check is a good idea. In general I'd prefer to use UTC as it the instance checking may not be in the same timezone and most servers (at least in the `*nix` world report their time in UTC):
```irb(main):004:0> Time.now.utc
=> 2018-05-03 14:59:46 UTC
# instead of
irb(main):002:0> Time.now
=> 2018-05-03 07:59:34 -0700```